docs/index.md

```diff
@@ -55,12 +55,12 @@ provider "netbox" {
 
 ### Required
 
+- `api_token` (String) Netbox API authentication token. Can be set via the `NETBOX_API_TOKEN` environment variable.
```

Comment:
Did you generate this doc with `go generate` in the root folder? I keep getting these diffs in PRs, but not on my machine.

Reply:
Yes, this change was from `go generate`.

I can revert it if you'd like, but technically `api_token` is a required attribute, correct?

```go
"api_token": {
Type:        schema.TypeString,
Required:    true,
DefaultFunc: schema.EnvDefaultFunc("NETBOX_API_TOKEN", nil),
Description: "Netbox API authentication token. Can be set via the `NETBOX_API_TOKEN` environment variable.",
},
```

Reply:
Yes, technically it is required, but the schema will render it as optional. There is an issue for that https://github.com/hashicorp/terraform-plugin-docs/issues/42 for that. I wonder how this can be inconsistent in our case, given that I put my fork in the `go generate` part. I also absolute cannot reproduce it, which drives me crazy :D

You can check with `terraform providers schema -json | jq '.provider_schemas."registry.terraform.io/e-breuninger/netbox".provider.block.attributes.api_token'` (the path might be slightly different for you) in a folder where the provider is `init`ed. You will see that the attribute is marked as `optional` there. At least it is for me :shrug: I tried it with tf 1.3.4 (latest) and 1.0.0, the results are the same.

Anyway, for the time being, just revert it, please. Same goes for the other MR.

Reply:
Interestingly enough, it shows as `required` for me:
```
$ terraform providers schema -json | jq '.provider_schemas."registry.terraform.io/e-breuninger/netbox".provider.block.attributes.api_token'
{
  "type": "string",
  "description": "Netbox API authentication token. Can be set via the `NETBOX_API_TOKEN` environment variable.",
  "description_kind": "markdown",
  "required": true
}

$ terraform version
Terraform v1.3.4
on linux_amd64
+ provider registry.terraform.io/e-breuninger/netbox v3.0.7
```

Either way, I reverted it for now.

Reply:
FYI, the output of the `terraform providers schema` function actually depends on the given environment variable. I have my NETBOX_API_TOKEN set to a value via bashrc and the function seems to detect that and determines that the value is now optional for me. If I unset it, it shows as required...

I updated the upstream docs.